app/services/controller.py

```diff
@@ -0,0 +1,284 @@
+from fastapi import FastAPI, APIRouter, HTTPException
+from app.database.schema import Channel
+from app.database.update_modal_channel import Change_channel
+from app.database.membership_modal import Membership
+from bson.objectid import ObjectId
+from app.database.db_queries import (
+    find_channel,
+    insert_channel,
+    find_membership,
+    insert_membership,
+    update_membership,
+    update_channelinfo,
+    delete_channel,
+    find_channels_all
+)
+from app.config.config import channel_DB, membership_DB
+from bson import ObjectId
+import datetime
+from app.services.error_custom import returnExceptions
+import pymongo
+
+
+
+
+def fetch_channels() -> list:
+    """Function to fetch all channels from the db
+
+    Returns:
+        channels[list]: [list of channel[dict]]
+    """
+    try:
+        query: dict = {"type": "public"}
+        channel_queried = find_channels_all(query)
+        if channel_queried is None:
+            raise returnExceptions(1008)
+        channels_list = []
+        for each_channel in channel_queried:
+            channels_list.append(Channel(**each_channel))
+        return channels_list
+    except pymongo.errors.PyMongoError as err:
+        raise returnExceptions(1004)
+
+
+def fetch_channel(id: str) -> dict:
+    """Function to fetch details of a channel with the given id
+
+    Args:
+        id (str): id created by the mongoDB
+
+    Returns:
+        channel_details[dict]: [the details of the channel requested]
+    """
+
+    try:
+        channel_details = find_channel(id)
+        if channel_details is None:
+            raise returnExceptions(1003)
+        else:
+            channel_details = dict(channel_details)
+            channel_details.pop("_id")
+            return channel_details
+    except pymongo.errors.PyMongoError as err:
+        raise returnExceptions(1004)
+
+
+def create_channel(user_id: str, channel: Channel) -> dict:
+    """Function to insert a channel into db
+
+    Args:
+        user_id[str] : [id of user requesting this service]
+        channel[dict] : [channel attributes like name,type,..]
+
+    Returns:
+        [dict]: [created channel is returned]
```

Comment:
1. remove square brackets from description
2. remove square brackets around `dict` in *line 74*